FrameView::frame should return an AbstractFrame

achristensen07 opened the pull request:
#### 6b38cc78aea225f0b479c89d29fe027e958635d0
<pre>
FrameView::frame should return an AbstractFrame
<a href="https://bugs.webkit.org/show_bug.cgi?id=249662">https://bugs.webkit.org/show_bug.cgi?id=249662</a>
rdar://103564957

Reviewed by Chris Dumez.

This is in preparation for a RemoteFrame to have a view object that is different
from FrameView but inherits from a shared parent class, like Frame/RemoteFrame/AbstractFrame
do now.

* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::appendAccessibilityObject):
(WebCore::AccessibilityObject::document const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::visiblePositionForPoint const):
(WebCore::AccessibilityRenderObject::remoteSVGRootElement const):
* Source/WebCore/accessibility/AccessibilityScrollView.cpp:
(WebCore::AccessibilityScrollView::document const):
* Source/WebCore/display/DisplayView.cpp:
(WebCore::Display::View::frame const):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::resolveStyle):
(WebCore::Document::attachToCachedFrame):
(WebCore::computeIntersectionState):
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::CaretBase::shouldRepaintCaret const):
* Source/WebCore/history/CachedFrame.cpp:
(WebCore::CachedFrameBase::restore):
(WebCore::CachedFrame::open):
(WebCore::CachedFrame::destroy):
* Source/WebCore/history/CachedPage.cpp:
(WebCore::CachedPage::restore):
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::InspectorInstrumentation::didAddOrRemoveScrollbarsImpl):
* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::didAddOrRemoveScrollbars):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::buildObjectForNode):
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
(WebCore::InspectorPageAgent::didPaint):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::subframeForTargetNode):
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::frameForWidget):
(WebCore::Frame::clearTimers):
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::~FrameView):
(WebCore::FrameView::removeFromAXObjectCache):
(WebCore::FrameView::init):
(WebCore::FrameView::recalculateScrollbarOverlayStyle):
(WebCore::FrameView::invalidateRect):
(WebCore::FrameView::setFrameRect):
(WebCore::FrameView::rootElementForCustomScrollbarPartStyle const):
(WebCore::FrameView::createScrollbar):
(WebCore::FrameView::setContentsSize):
(WebCore::FrameView::adjustViewSize):
(WebCore::FrameView::applyOverflowToViewport):
(WebCore::FrameView::applyPaginationToViewport):
(WebCore::FrameView::calculateScrollbarModesForLayout):
(WebCore::FrameView::updateSnapOffsets):
(WebCore::FrameView::flushCompositingStateForThisFrame):
(WebCore::FrameView::setNeedsOneShotDrawingSynchronization):
(WebCore::FrameView::graphicsLayerForPageScale):
(WebCore::FrameView::graphicsLayerForTransientZoomShadow):
(WebCore::FrameView::fixedScrollableAreaBoundsInflatedForScrolling const):
(WebCore::FrameView::scrollPositionRespectingCustomFixedPosition const):
(WebCore::FrameView::headerHeight const):
(WebCore::FrameView::footerHeight const):
(WebCore::FrameView::topContentInset const):
(WebCore::FrameView::isEnclosedInCompositingLayer const):
(WebCore::FrameView::flushCompositingStateIncludingSubframes):
(WebCore::FrameView::forceLayoutParentViewIfNeeded):
(WebCore::FrameView::markRootOrBodyRendererDirty const):
(WebCore::FrameView::adjustScrollbarsForLayout):
(WebCore::FrameView::willDoLayout):
(WebCore::FrameView::didLayout):
(WebCore::FrameView::mediaType const):
(WebCore::FrameView::mockScrollbarsControllerEnabled const):
(WebCore::FrameView::logMockScrollbarsControllerMessage const):
(WebCore::FrameView::debugDescription const):
(WebCore::FrameView::setBaseLayoutViewportOrigin):
(WebCore::FrameView::setLayoutViewportOverrideRect):
(WebCore::FrameView::updateLayoutViewport):
(WebCore::FrameView::viewportConstrainedVisibleContentRect const):
(WebCore::FrameView::rectForFixedPositionLayout const):
(WebCore::FrameView::frameScaleFactor const):
(WebCore::FrameView::scrollPositionForFixedPosition const):
(WebCore::FrameView::viewportConstrainedObjectsRect const):
(WebCore::FrameView::minimumScrollPosition const):
(WebCore::FrameView::maximumScrollPosition const):
(WebCore::FrameView::unscaledMinimumScrollPosition const):
(WebCore::FrameView::unscaledMaximumScrollPosition const):
(WebCore::FrameView::viewportContentsChanged):
(WebCore::FrameView::viewRectExpandedByContentInsets const):
(WebCore::FrameView::fixedElementsLayoutRelativeToFrame const):
(WebCore::FrameView::lastKnownMousePositionInView const):
(WebCore::FrameView::isHandlingWheelEvent const):
(WebCore::FrameView::shouldSetCursor const):
(WebCore::FrameView::rendererForColorScheme const):
(WebCore::FrameView::useDarkAppearance const):
(WebCore::FrameView::styleColorOptions const):
(WebCore::FrameView::scrollContentsFastPath):
(WebCore::FrameView::scrollContentsSlowPath):
(WebCore::FrameView::scrollToFragment):
(WebCore::FrameView::scrollToFragmentInternal):
(WebCore::FrameView::maintainScrollPositionAtAnchor):
(WebCore::FrameView::scrollElementToRect):
(WebCore::FrameView::setScrollPosition):
(WebCore::FrameView::resetScrollAnchor):
(WebCore::FrameView::scrollToFocusedElementInternal):
(WebCore::FrameView::textFragmentIndicatorTimerFired):
(WebCore::FrameView::scrollRectToVisible):
(WebCore::scrollPositionChangeOptionsForElement):
(WebCore::FrameView::scrollRectToVisibleInChildView):
(WebCore::FrameView::scrollRectToVisibleInTopLevelView):
(WebCore::FrameView::setFixedVisibleContentRect):
(WebCore::FrameView::didChangeScrollOffset):
(WebCore::FrameView::scrollOffsetChangedViaPlatformWidgetImpl):
(WebCore::FrameView::scrollPositionChanged):
(WebCore::FrameView::applyRecursivelyWithVisibleRect):
(WebCore::FrameView::resumeVisibleImageAnimations):
(WebCore::FrameView::updatePlayStateForAllAnimations):
(WebCore::FrameView::updateScriptedAnimationsAndTimersThrottlingState):
(WebCore::FrameView::scrollingCoordinator const):
(WebCore::FrameView::shouldUpdateCompositingLayersAfterScrolling const):
(WebCore::FrameView::hostWindow const):
(WebCore::FrameView::repaintContentRectangle):
(WebCore::FrameView::renderedCharactersExceed):
(WebCore::FrameView::availableContentSizeChanged):
(WebCore::FrameView::updateContentsSize):
(WebCore::FrameView::computeScrollability const):
(WebCore::FrameView::layoutOrVisualViewportChanged):
(WebCore::FrameView::loadProgressingStatusChanged):
(WebCore::shouldEnableSpeculativeTilingDuringLoading):
(WebCore::FrameView::show):
(WebCore::FrameView::calculateExtendedBackgroundMode const):
(WebCore::FrameView::safeToPropagateScrollToParent const):
(WebCore::FrameView::scrollToAnchor):
(WebCore::FrameView::scrollToTextFragmentRange):
(WebCore::FrameView::performPostLayoutTasks):
(WebCore::FrameView::scheduleResizeEventIfNeeded):
(WebCore::FrameView::autoSizeIfEnabled):
(WebCore::FrameView::performFixedWidthAutoSize):
(WebCore::FrameView::performSizeToContentAutoSize):
(WebCore::FrameView::viewportRenderer const):
(WebCore::FrameView::updateOverflowStatus):
(WebCore::FrameView::pagination const):
(WebCore::FrameView::setPagination):
(WebCore::FrameView::windowClipRect const):
(WebCore::FrameView::isActive const):
(WebCore::FrameView::forceUpdateScrollbarsOnMainThreadForPerformanceTesting const):
(WebCore::FrameView::adjustVerticalPageScrollStepForFixedContent):
(WebCore::FrameView::visibleContentScaleFactor const):
(WebCore::FrameView::setVisibleScrollerThumbRect):
(WebCore::FrameView::enclosingScrollableArea const):
(WebCore::FrameView::scrollableAreaBoundingBox const):
(WebCore::FrameView::isScrollable):
(WebCore::FrameView::hasScrollableOrRubberbandableAncestor):
(WebCore::FrameView::shouldSuspendScrollAnimations const):
(WebCore::FrameView::scrollbarStyleChanged):
(WebCore::FrameView::notifyAllFramesThatContentAreaWillPaint const):
(WebCore::FrameView::notifyScrollableAreasThatContentAreaWillPaint const):
(WebCore::FrameView::scrollAnimatorEnabled const):
(WebCore::FrameView::updateScrollCorner):
(WebCore::FrameView::paintScrollCorner):
(WebCore::FrameView::paintScrollbar):
(WebCore::FrameView::documentBackgroundColor const):
(WebCore::FrameView::parentFrameView const):
(WebCore::FrameView::updateControlTints):
(WebCore::FrameView::willPaintContents):
(WebCore::FrameView::paintContents):
(WebCore::FrameView::paintOverhangAreas):
(WebCore::FrameView::updateLayoutAndStyleIfNeededRecursive):
(WebCore::FrameView::updateHasReachedSignificantRenderedTextThreshold):
(WebCore::FrameView::qualifiesAsSignificantRenderedText const):
(WebCore::FrameView::checkAndDispatchDidReachVisuallyNonEmptyState):
(WebCore::FrameView::enableSizeToContentAutoSizeMode):
(WebCore::FrameView::forceLayoutForPagination):
(WebCore::FrameView::convertToContainingView const):
(WebCore::FrameView::convertFromContainingView const):
(WebCore::FrameView::documentToAbsoluteScaleFactor const):
(WebCore::FrameView::documentToClientOffset const):
(WebCore::FrameView::absoluteToLayoutViewportPoint const):
(WebCore::FrameView::layoutViewportToAbsolutePoint const):
(WebCore::FrameView::layoutViewportToAbsoluteRect const):
(WebCore::FrameView::absoluteToLayoutViewportRect const):
(WebCore::FrameView::clientToLayoutViewportRect const):
(WebCore::FrameView::clientToLayoutViewportPoint const):
(WebCore::FrameView::setTracksRepaints):
(WebCore::FrameView::trackedRepaintRectsAsText const):
(WebCore::FrameView::scheduleScrollEvent):
(WebCore::FrameView::axObjectCache const):
(WebCore::FrameView::useCustomFixedPositionLayoutRect const):
(WebCore::FrameView::updateFixedPositionLayoutRect):
(WebCore::FrameView::setScrollingPerformanceTestingEnabled):
(WebCore::FrameView::didAddScrollbar):
(WebCore::FrameView::fireLayoutRelatedMilestonesIfNeeded):
(WebCore::FrameView::firePaintRelatedMilestonesIfNeeded):
(WebCore::FrameView::setVisualUpdatesAllowedByClient):
(WebCore::FrameView::scrollBehaviorForFixedElements const):
(WebCore::FrameView::renderView const):
(WebCore::FrameView::mapFromLayoutToCSSUnits const):
(WebCore::FrameView::mapFromCSSToLayoutUnits const):
(WebCore::FrameView::setViewExposedRect):
(WebCore::FrameView::clearSizeOverrideForCSSDefaultViewportUnits):
(WebCore::FrameView::setOverrideSizeForCSSDefaultViewportUnits):
(WebCore::FrameView::clearSizeOverrideForCSSSmallViewportUnits):
(WebCore::FrameView::setOverrideSizeForCSSSmallViewportUnits):
(WebCore::FrameView::clearSizeOverrideForCSSLargeViewportUnits):
(WebCore::FrameView::setOverrideSizeForCSSLargeViewportUnits):
(WebCore::FrameView::sizeForCSSDynamicViewportUnits const):
(WebCore::FrameView::didFinishProhibitingScrollingWhenChangingContentSize):
(WebCore::FrameView::pageScaleFactor const):
(WebCore::FrameView::didStartScrollAnimation):
(WebCore::FrameView::updateScrollbarSteps):
(WebCore::FrameView::horizontalOverscrollBehavior const):
(WebCore::FrameView::verticalOverscrollBehavior const):
(WebCore::FrameView::isVisibleToHitTesting const):
(WebCore::FrameView::getPossiblyFixedRectToExpose const):
* Source/WebCore/page/FrameView.h:
* Source/WebCore/page/FrameViewLayoutContext.cpp:
(WebCore::FrameViewLayoutContext::performLayout):
(WebCore::FrameViewLayoutContext::frame const):
* Source/WebCore/page/ModalContainerObserver.cpp:
(WebCore::ModalContainerObserver::updateModalContainerIfNeeded):
(WebCore::ModalContainerObserver::hideUserInteractionBlockingElementIfNeeded):
* Source/WebCore/page/PageColorSampler.cpp:
(WebCore::sampleColor):
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::passWheelEventToWidget):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
(WebCore::AsyncScrollingCoordinator::frameViewEventTrackingRegionsChanged):
(WebCore::AsyncScrollingCoordinator::requestScrollPositionUpdate):
(WebCore::AsyncScrollingCoordinator::updateScrollPositionAfterAsyncScroll):
(WebCore::AsyncScrollingCoordinator::ensureRootStateNodeForFrameView):
(WebCore::AsyncScrollingCoordinator::setFrameScrollingNodeState):
* Source/WebCore/page/scrolling/ScrollingCoordinator.cpp:
(WebCore::ScrollingCoordinator::coordinatesScrollingForFrameView const):
(WebCore::ScrollingCoordinator::scrollContainerLayerForFrameView):
(WebCore::ScrollingCoordinator::scrolledContentsLayerForFrameView):
(WebCore::ScrollingCoordinator::headerLayerForFrameView):
(WebCore::ScrollingCoordinator::footerLayerForFrameView):
(WebCore::ScrollingCoordinator::counterScrollingLayerForFrameView):
(WebCore::ScrollingCoordinator::insetClipLayerForFrameView):
(WebCore::ScrollingCoordinator::contentShadowLayerForFrameView):
(WebCore::ScrollingCoordinator::rootContentsLayerForFrameView):
(WebCore::ScrollingCoordinator::updateSynchronousScrollingReasons):
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::adjustComputedFontSizesOnBlocks):
(WebCore::RenderElement::resetTextAutosizing):
* Source/WebCore/rendering/RenderIFrame.cpp:
(WebCore::RenderIFrame::contentRootRenderer const):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::hitTest):
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::isMainFrameCompositor const):
(WebCore::RenderLayerCompositor::ensureRootLayer):
(WebCore::RenderLayerCompositor::attachRootLayer):
(WebCore::RenderLayerCompositor::detachRootLayer):
(WebCore::RenderLayerCompositor::rootLayerAttachmentChanged):
* Source/WebCore/rendering/RenderTreeAsText.cpp:
(WebCore::write):
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::clientLogicalWidthForFixedPosition const):
(WebCore::RenderView::clientLogicalHeightForFixedPosition const):
(WebCore::RenderView::shouldUsePrintingLayout const):
(WebCore::RenderView::zoomFactor const):
* Source/WebCore/rendering/RenderWidget.cpp:
(WebCore::RenderWidget::updateWidgetPosition):
(WebCore::RenderWidget::nodeAtPoint):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::convertRectFromFrameClientToRootView):
(WebKit::convertPointFromFrameClientToRootView):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::shouldUseTiledBackingForFrameView const):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::dynamicViewportSizeUpdate):
(WebKit::WebPage::updateVisibleContentRects):
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::shouldUseTiledBackingForFrameView const):

Canonical link: <a href="https://commits.webkit.org/258209@main">https://commits.webkit.org/258209@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9c4a47fba0d4b2a63b5458a28c9e29df91d1e1f3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101223 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10376 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34276 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/110508 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/170793 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/105210 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11313 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1243 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/93628 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/108342 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107006 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/8609 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/91851 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/93628 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/90511 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/23257 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/93628 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4017 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/24771 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/4067 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/1199 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/10172 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44248 "Passed tests") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/5821 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2961 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/FrameView.cpp:
```diff
@@ -253,7 +253,7 @@ FrameView::~FrameView()
     ASSERT(!m_scrollCorner);
 
-    ASSERT(frame().view() != this || !frame().contentRenderer());
+    ASSERT(m_frame.get().view() != this || !m_frame.get().contentRenderer());
```
Ugh, all these `m_frame.get().` in this file should be `m_frame->`.

achristensen07 replied:
I like m_frame.get(). better because it makes it more clear that nothing can be null, whereas m_frame-> looks like it might be a RefPtr.  I made this suggested change, though.

cdumez replied:
`m_frame.get().` : m_frame could still be a RefPtr.

achristensen07 replied:
It would need to be m_frame.get()->

cdumez reviewed (approved):
LGTM assuming the bots are happy.

achristensen07 committed:
FrameView::frame should return an AbstractFrame
https://bugs.webkit.org/show_bug.cgi?id=249662
rdar://103564957

Reviewed by Chris Dumez.

This is in preparation for a RemoteFrame to have a view object that is different
from FrameView but inherits from a shared parent class, like Frame/RemoteFrame/AbstractFrame
do now.

* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::appendAccessibilityObject):
(WebCore::AccessibilityObject::document const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::visiblePositionForPoint const):
(WebCore::AccessibilityRenderObject::remoteSVGRootElement const):
* Source/WebCore/accessibility/AccessibilityScrollView.cpp:
(WebCore::AccessibilityScrollView::document const):
* Source/WebCore/display/DisplayView.cpp:
(WebCore::Display::View::frame const):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::resolveStyle):
(WebCore::Document::attachToCachedFrame):
(WebCore::computeIntersectionState):
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::CaretBase::shouldRepaintCaret const):
* Source/WebCore/history/CachedFrame.cpp:
(WebCore::CachedFrameBase::restore):
(WebCore::CachedFrame::open):
(WebCore::CachedFrame::destroy):
* Source/WebCore/history/CachedPage.cpp:
(WebCore::CachedPage::restore):
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::InspectorInstrumentation::didAddOrRemoveScrollbarsImpl):
* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::didAddOrRemoveScrollbars):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::buildObjectForNode):
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
(WebCore::InspectorPageAgent::didPaint):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::subframeForTargetNode):
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::frameForWidget):
(WebCore::Frame::clearTimers):
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::~FrameView):
(WebCore::FrameView::removeFromAXObjectCache):
(WebCore::FrameView::init):
(WebCore::FrameView::recalculateScrollbarOverlayStyle):
(WebCore::FrameView::invalidateRect):
(WebCore::FrameView::setFrameRect):
(WebCore::FrameView::rootElementForCustomScrollbarPartStyle const):
(WebCore::FrameView::createScrollbar):
(WebCore::FrameView::setContentsSize):
(WebCore::FrameView::adjustViewSize):
(WebCore::FrameView::applyOverflowToViewport):
(WebCore::FrameView::applyPaginationToViewport):
(WebCore::FrameView::calculateScrollbarModesForLayout):
(WebCore::FrameView::updateSnapOffsets):
(WebCore::FrameView::flushCompositingStateForThisFrame):
(WebCore::FrameView::setNeedsOneShotDrawingSynchronization):
(WebCore::FrameView::graphicsLayerForPageScale):
(WebCore::FrameView::graphicsLayerForTransientZoomShadow):
(WebCore::FrameView::fixedScrollableAreaBoundsInflatedForScrolling const):
(WebCore::FrameView::scrollPositionRespectingCustomFixedPosition const):
(WebCore::FrameView::headerHeight const):
(WebCore::FrameView::footerHeight const):
(WebCore::FrameView::topContentInset const):
(WebCore::FrameView::isEnclosedInCompositingLayer const):
(WebCore::FrameView::flushCompositingStateIncludingSubframes):
(WebCore::FrameView::forceLayoutParentViewIfNeeded):
(WebCore::FrameView::markRootOrBodyRendererDirty const):
(WebCore::FrameView::adjustScrollbarsForLayout):
(WebCore::FrameView::willDoLayout):
(WebCore::FrameView::didLayout):
(WebCore::FrameView::mediaType const):
(WebCore::FrameView::mockScrollbarsControllerEnabled const):
(WebCore::FrameView::logMockScrollbarsControllerMessage const):
(WebCore::FrameView::debugDescription const):
(WebCore::FrameView::setBaseLayoutViewportOrigin):
(WebCore::FrameView::setLayoutViewportOverrideRect):
(WebCore::FrameView::updateLayoutViewport):
(WebCore::FrameView::viewportConstrainedVisibleContentRect const):
(WebCore::FrameView::rectForFixedPositionLayout const):
(WebCore::FrameView::frameScaleFactor const):
(WebCore::FrameView::scrollPositionForFixedPosition const):
(WebCore::FrameView::viewportConstrainedObjectsRect const):
(WebCore::FrameView::minimumScrollPosition const):
(WebCore::FrameView::maximumScrollPosition const):
(WebCore::FrameView::unscaledMinimumScrollPosition const):
(WebCore::FrameView::unscaledMaximumScrollPosition const):
(WebCore::FrameView::viewportContentsChanged):
(WebCore::FrameView::viewRectExpandedByContentInsets const):
(WebCore::FrameView::fixedElementsLayoutRelativeToFrame const):
(WebCore::FrameView::lastKnownMousePositionInView const):
(WebCore::FrameView::isHandlingWheelEvent const):
(WebCore::FrameView::shouldSetCursor const):
(WebCore::FrameView::rendererForColorScheme const):
(WebCore::FrameView::useDarkAppearance const):
(WebCore::FrameView::styleColorOptions const):
(WebCore::FrameView::scrollContentsFastPath):
(WebCore::FrameView::scrollContentsSlowPath):
(WebCore::FrameView::scrollToFragment):
(WebCore::FrameView::scrollToFragmentInternal):
(WebCore::FrameView::maintainScrollPositionAtAnchor):
(WebCore::FrameView::scrollElementToRect):
(WebCore::FrameView::setScrollPosition):
(WebCore::FrameView::resetScrollAnchor):
(WebCore::FrameView::scrollToFocusedElementInternal):
(WebCore::FrameView::textFragmentIndicatorTimerFired):
(WebCore::FrameView::scrollRectToVisible):
(WebCore::scrollPositionChangeOptionsForElement):
(WebCore::FrameView::scrollRectToVisibleInChildView):
(WebCore::FrameView::scrollRectToVisibleInTopLevelView):
(WebCore::FrameView::setFixedVisibleContentRect):
(WebCore::FrameView::didChangeScrollOffset):
(WebCore::FrameView::scrollOffsetChangedViaPlatformWidgetImpl):
(WebCore::FrameView::scrollPositionChanged):
(WebCore::FrameView::applyRecursivelyWithVisibleRect):
(WebCore::FrameView::resumeVisibleImageAnimations):
(WebCore::FrameView::updatePlayStateForAllAnimations):
(WebCore::FrameView::updateScriptedAnimationsAndTimersThrottlingState):
(WebCore::FrameView::scrollingCoordinator const):
(WebCore::FrameView::shouldUpdateCompositingLayersAfterScrolling const):
(WebCore::FrameView::hostWindow const):
(WebCore::FrameView::repaintContentRectangle):
(WebCore::FrameView::renderedCharactersExceed):
(WebCore::FrameView::availableContentSizeChanged):
(WebCore::FrameView::updateContentsSize):
(WebCore::FrameView::computeScrollability const):
(WebCore::FrameView::layoutOrVisualViewportChanged):
(WebCore::FrameView::loadProgressingStatusChanged):
(WebCore::shouldEnableSpeculativeTilingDuringLoading):
(WebCore::FrameView::show):
(WebCore::FrameView::calculateExtendedBackgroundMode const):
(WebCore::FrameView::safeToPropagateScrollToParent const):
(WebCore::FrameView::scrollToAnchor):
(WebCore::FrameView::scrollToTextFragmentRange):
(WebCore::FrameView::performPostLayoutTasks):
(WebCore::FrameView::scheduleResizeEventIfNeeded):
(WebCore::FrameView::autoSizeIfEnabled):
(WebCore::FrameView::performFixedWidthAutoSize):
(WebCore::FrameView::performSizeToContentAutoSize):
(WebCore::FrameView::viewportRenderer const):
(WebCore::FrameView::updateOverflowStatus):
(WebCore::FrameView::pagination const):
(WebCore::FrameView::setPagination):
(WebCore::FrameView::windowClipRect const):
(WebCore::FrameView::isActive const):
(WebCore::FrameView::forceUpdateScrollbarsOnMainThreadForPerformanceTesting const):
(WebCore::FrameView::adjustVerticalPageScrollStepForFixedContent):
(WebCore::FrameView::visibleContentScaleFactor const):
(WebCore::FrameView::setVisibleScrollerThumbRect):
(WebCore::FrameView::enclosingScrollableArea const):
(WebCore::FrameView::scrollableAreaBoundingBox const):
(WebCore::FrameView::isScrollable):
(WebCore::FrameView::hasScrollableOrRubberbandableAncestor):
(WebCore::FrameView::shouldSuspendScrollAnimations const):
(WebCore::FrameView::scrollbarStyleChanged):
(WebCore::FrameView::notifyAllFramesThatContentAreaWillPaint const):
(WebCore::FrameView::notifyScrollableAreasThatContentAreaWillPaint const):
(WebCore::FrameView::scrollAnimatorEnabled const):
(WebCore::FrameView::updateScrollCorner):
(WebCore::FrameView::paintScrollCorner):
(WebCore::FrameView::paintScrollbar):
(WebCore::FrameView::documentBackgroundColor const):
(WebCore::FrameView::parentFrameView const):
(WebCore::FrameView::updateControlTints):
(WebCore::FrameView::willPaintContents):
(WebCore::FrameView::paintContents):
(WebCore::FrameView::paintOverhangAreas):
(WebCore::FrameView::updateLayoutAndStyleIfNeededRecursive):
(WebCore::FrameView::updateHasReachedSignificantRenderedTextThreshold):
(WebCore::FrameView::qualifiesAsSignificantRenderedText const):
(WebCore::FrameView::checkAndDispatchDidReachVisuallyNonEmptyState):
(WebCore::FrameView::enableSizeToContentAutoSizeMode):
(WebCore::FrameView::forceLayoutForPagination):
(WebCore::FrameView::convertToContainingView const):
(WebCore::FrameView::convertFromContainingView const):
(WebCore::FrameView::documentToAbsoluteScaleFactor const):
(WebCore::FrameView::documentToClientOffset const):
(WebCore::FrameView::absoluteToLayoutViewportPoint const):
(WebCore::FrameView::layoutViewportToAbsolutePoint const):
(WebCore::FrameView::layoutViewportToAbsoluteRect const):
(WebCore::FrameView::absoluteToLayoutViewportRect const):
(WebCore::FrameView::clientToLayoutViewportRect const):
(WebCore::FrameView::clientToLayoutViewportPoint const):
(WebCore::FrameView::setTracksRepaints):
(WebCore::FrameView::trackedRepaintRectsAsText const):
(WebCore::FrameView::scheduleScrollEvent):
(WebCore::FrameView::axObjectCache const):
(WebCore::FrameView::useCustomFixedPositionLayoutRect const):
(WebCore::FrameView::updateFixedPositionLayoutRect):
(WebCore::FrameView::setScrollingPerformanceTestingEnabled):
(WebCore::FrameView::didAddScrollbar):
(WebCore::FrameView::fireLayoutRelatedMilestonesIfNeeded):
(WebCore::FrameView::firePaintRelatedMilestonesIfNeeded):
(WebCore::FrameView::setVisualUpdatesAllowedByClient):
(WebCore::FrameView::scrollBehaviorForFixedElements const):
(WebCore::FrameView::renderView const):
(WebCore::FrameView::mapFromLayoutToCSSUnits const):
(WebCore::FrameView::mapFromCSSToLayoutUnits const):
(WebCore::FrameView::setViewExposedRect):
(WebCore::FrameView::clearSizeOverrideForCSSDefaultViewportUnits):
(WebCore::FrameView::setOverrideSizeForCSSDefaultViewportUnits):
(WebCore::FrameView::clearSizeOverrideForCSSSmallViewportUnits):
(WebCore::FrameView::setOverrideSizeForCSSSmallViewportUnits):
(WebCore::FrameView::clearSizeOverrideForCSSLargeViewportUnits):
(WebCore::FrameView::setOverrideSizeForCSSLargeViewportUnits):
(WebCore::FrameView::sizeForCSSDynamicViewportUnits const):
(WebCore::FrameView::didFinishProhibitingScrollingWhenChangingContentSize):
(WebCore::FrameView::pageScaleFactor const):
(WebCore::FrameView::didStartScrollAnimation):
(WebCore::FrameView::updateScrollbarSteps):
(WebCore::FrameView::horizontalOverscrollBehavior const):
(WebCore::FrameView::verticalOverscrollBehavior const):
(WebCore::FrameView::isVisibleToHitTesting const):
(WebCore::FrameView::getPossiblyFixedRectToExpose const):
* Source/WebCore/page/FrameView.h:
* Source/WebCore/page/FrameViewLayoutContext.cpp:
(WebCore::FrameViewLayoutContext::performLayout):
(WebCore::FrameViewLayoutContext::frame const):
* Source/WebCore/page/ModalContainerObserver.cpp:
(WebCore::ModalContainerObserver::updateModalContainerIfNeeded):
(WebCore::ModalContainerObserver::hideUserInteractionBlockingElementIfNeeded):
* Source/WebCore/page/PageColorSampler.cpp:
(WebCore::sampleColor):
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::passWheelEventToWidget):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
(WebCore::AsyncScrollingCoordinator::frameViewEventTrackingRegionsChanged):
(WebCore::AsyncScrollingCoordinator::requestScrollPositionUpdate):
(WebCore::AsyncScrollingCoordinator::updateScrollPositionAfterAsyncScroll):
(WebCore::AsyncScrollingCoordinator::ensureRootStateNodeForFrameView):
(WebCore::AsyncScrollingCoordinator::setFrameScrollingNodeState):
* Source/WebCore/page/scrolling/ScrollingCoordinator.cpp:
(WebCore::ScrollingCoordinator::coordinatesScrollingForFrameView const):
(WebCore::ScrollingCoordinator::scrollContainerLayerForFrameView):
(WebCore::ScrollingCoordinator::scrolledContentsLayerForFrameView):
(WebCore::ScrollingCoordinator::headerLayerForFrameView):
(WebCore::ScrollingCoordinator::footerLayerForFrameView):
(WebCore::ScrollingCoordinator::counterScrollingLayerForFrameView):
(WebCore::ScrollingCoordinator::insetClipLayerForFrameView):
(WebCore::ScrollingCoordinator::contentShadowLayerForFrameView):
(WebCore::ScrollingCoordinator::rootContentsLayerForFrameView):
(WebCore::ScrollingCoordinator::updateSynchronousScrollingReasons):
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::adjustComputedFontSizesOnBlocks):
(WebCore::RenderElement::resetTextAutosizing):
* Source/WebCore/rendering/RenderIFrame.cpp:
(WebCore::RenderIFrame::contentRootRenderer const):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::hitTest):
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::isMainFrameCompositor const):
(WebCore::RenderLayerCompositor::ensureRootLayer):
(WebCore::RenderLayerCompositor::attachRootLayer):
(WebCore::RenderLayerCompositor::detachRootLayer):
(WebCore::RenderLayerCompositor::rootLayerAttachmentChanged):
* Source/WebCore/rendering/RenderTreeAsText.cpp:
(WebCore::write):
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::clientLogicalWidthForFixedPosition const):
(WebCore::RenderView::clientLogicalHeightForFixedPosition const):
(WebCore::RenderView::shouldUsePrintingLayout const):
(WebCore::RenderView::zoomFactor const):
* Source/WebCore/rendering/RenderWidget.cpp:
(WebCore::RenderWidget::updateWidgetPosition):
(WebCore::RenderWidget::nodeAtPoint):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::convertRectFromFrameClientToRootView):
(WebKit::convertPointFromFrameClientToRootView):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::shouldUseTiledBackingForFrameView const):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::dynamicViewportSizeUpdate):
(WebKit::WebPage::updateVisibleContentRects):
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::shouldUseTiledBackingForFrameView const):

Canonical link: https://commits.webkit.org/258209@main
